pkg/operator/operator.go

```diff
@@ -226,6 +230,40 @@ func isOwned(obj interface{}) (bool, error) {
 	return ok, nil
 }
 
+func (optr *Operator) eventHandlerSingleton() cache.FilteringResourceEventHandler {
+	workQueueKey := fmt.Sprintf("%s/%s", optr.namespace, optr.name)
```

Comment:
this seems fine. I was referring though if we could somehow leverage the built in library e.g `tweakListOptions internalinterfaces.TweakListOptionsFunc` but seems not possible without creating renouncing to the build factory, e.g `NewFilteredDeploymentInformer()`